filefinder/_filefinder.py

```diff
@@ -135,21 +141,32 @@ def _glob(pattern):
 
         return glob.glob(pattern)
 
-    def _parse_paths(self, paths):
+    def _parse_paths(self, paths, on_parse_error):
 
         out = list()
         for path in paths:
             parsed = self.parser.parse(path)
 
             if parsed is None:
-                raise ValueError(
-                    f"Could not parse '{path}' with the pattern '{self.pattern}' - are"
-                    " there contradictory values?"
-                )
-
-            out.append([path + self._suffix] + list(parsed.named.values()))
+                if on_parse_error == "raise":
+                    raise ValueError(
+                        f"Could not parse '{path}' with the pattern '{self.pattern}' - are"
+                        " there contradictory values?"
+                    )
+                elif on_parse_error == "warn":
+                    warnings.warn(
+                        f"Could not parse '{path}' with the pattern '{self.pattern}' - are"
+                        " there contradictory values?"
+                    )
+                elif on_parse_error == "skip":
+                    pass
+            else:
+                out.append([path + self._suffix] + list(parsed.named.values()))
 
-        keys = ["filename"] + list(parsed.named.keys())
+        if out == list():
```

Comment:
Thinking a bit more about this: it will be difficult to cover all cases, so we are better off replacing this whole block with:


```python
keys = ["filename"] + sorted(self.keys)
```
(i.e. my original approach was not optimal)

Reply:
But using my suggestion will sort the keys differently - fine to update the failing tests for now.

Reply:
If `on_parse_error` is not one of raise, warn skip the behavior is the same as in skip, this is the same as the on_missing argument in https://github.com/mathause/filefinder/blob/main/filefinder/filters.py I think. Should I raise an error if it is not one of the options?

Reply:
> But using my suggestion will sort the keys differently - fine to update the failing tests for now.

This doesn't work because it just resorts the keys but not the data entries, so they are labelled incorrectly. Also, just `list(self.keys)` does not work. I don't really understand how the keys are sorted in self and in parsed.

Reply:
> If `on_parse_error` is not one of raise, warn skip the behavior is the same as in skip, this is the same as the on_missing argument in https://github.com/mathause/filefinder/blob/main/filefinder/filters.py I think. Should I raise an error if it is not one of the options?

Good idea

Reply:
> > But using my suggestion will sort the keys differently - fine to update the failing tests for now.
> 
> This doesn't work because it just resorts the keys but not the data entries, so they are labelled incorrectly. Also, just `list(self.keys)` does not work. I don't really understand how the keys are sorted in self and in parsed.

Yes, it's a `set` so the ordering random. If you call `sorted` on it it will sort it alphabetically.

Reply:
Oh right, sorry - just read your post correctly. Let me think about this...

Reply:
Create unsorted unique tuples:

```python
keys = tuple(pd.Series(keys).unique())
```